internal/pkg/deploy/pipeline.go

```diff
@@ -442,41 +445,143 @@ func (s *GitHubV1Source) Owner() (string, error) {
 	return owner, nil
 }
 
+type associatedEnvironment struct {
+	// Name of the environment, must be unique within an application.
+	// This is also the name of the pipeline stage.
+	Name string
+
+	// The region this environment is created in.
+	Region string
+
+	// AppName represents the application name the environment is part of.
+	AppName string
```

Comment:
Is `AppName` always the app where the pipeline lives in?

Reply:
Yeah! Did you have a concern or a refactor suggestion for it?

Reply:
My concern was that it might be a piece of redundant information (there is `.AppName` as well at the top-level `CreatePipelineInput`). This means it may be possible that at some point in the future, by some accidents maybe, this `AppName` holds a different value  than `CreatePipelineInput.AppName` which will cause a bug.

I see why it's needed here, and how it plays a role in the `DeployAction` interface design. I don't have a good refactor suggestion though : (